Sources/Subs.php

```diff
@@ -3704,7 +3704,7 @@ function getAttachmentFilename($filename, $attachment_id, $dir = null, $new = fa
 		$file_hash = sha1(md5($filename . time()) . mt_rand());
 
 	// Are we using multiple directories?
-	if (!empty($modSettings['currentAttachmentUploadDir']))
+	if (is_array($modSettings['attachmentUploadDir']))
```

Comment:
In theory, `!empty($modSettings['currentAttachmentUploadDir'])` will always be the same as `is_array($modSettings['attachmentUploadDir'])`. But I suppose a mod or something could mess with the value of `$modSettings['currentAttachmentUploadDir']`, so this change makes sense.